### PR TITLE
First draft of search select.

### PR DESCRIPTION
**Pull request by umamialex**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+AmpersandCalendar
```
Comment (nubs) [inline README.md]: this should be changed :)

File: ampersand-search-select.js [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+    },
+    session: {
+      query: [ 'string', false, ''],
+      data: 'object',
+      idAttribute: 'string',
+      imageAttribute: 'string',
+      queryAttribute: 'string',
+
+      // Private Variables
+      _data: 'object',
```
Comment (nubs) [inline ampersand-search-select.js:21]: why put these in `session`?  Why not just stick them directly on the object?

File: ampersand-search-select.js [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+      _querySet: [ 'array', false, function() { return []; } ]
+    },
+    initialize: function() {
+      this._data = new AmpersandSubCollection(this.data);
+      this.selected = new AmpersandSubCollection(this.data);
+
+      this.on('change:_selected', function(selectState, selected) {
+        selectState.selected.configure({
+          filter: function(model) {
+            return !!~selected.indexOf(model[selectState.idAttribute]);
```
Comment (nubs) [inline ampersand-search-select.js]: Can you use `!== -1` instead of `!!~`?  Combining numbers, bitwise operators (involving sign flipping and two's complement) and boolean operators into a single expression is probably best avoided. Except maybe for code golf, but `n+1` beats `!!~n`, so this seems more like obfuscation :p

File: ampersand-search-select.js [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+
+      searchGlass.append('line')
+        .attr('class', 'ampersand-search-select-glass-line')
+        .attr('x1', 0)
+        .attr('y1', 17)
+        .attr('x2', 7.5)
+        .attr('y2', 10);
+    },
+    renderQuerySet: function(querySet) {
+      if (this.svg) {
```
Comment (nubs) [inline ampersand-search-select.js]: can you reverse this condition and just return early?  It's hard to follow long spans of indentation like this.

File: package.json [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  "name": "ampersand-search-select",
+  "version": "0.1.0",
+  "description": "Creates a search bar that shows a list of matching items that can be selected.",
+  "main": "ampersand-search-select.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git@github.com:helpdotcom/ampersand-search-select"
```
Comment (nubs) [inline package.json]: can you switch this to a public url - http:// or git:// protocol?

File: ampersand-search-select.js (hunk 2 above)
Comment (umamialex) [inline ampersand-search-select.js:21]: Sticking to the idea that `.toJSON()` should only give you the stuff you should care about. I feel like the other thing you might care about is the `query`, but everything else besides the private variables are things the user sets.

**Comment by umamialex** [comment]
Reupped.

File: ampersand-search-select.js (hunk 2 above)
Comment (nubs) [inline ampersand-search-select.js:21]: Sorry, I didn't mean sticking it in `props`, but sticking it directly on the object like:

``` js
var SearchSelectState = AmpersandState.extend({
  _data: {}
});
```

Comment (nubs) [inline ampersand-search-select.js:21]: ping

**Comment by nubs** [comment]
Talked about things in person, and nothing is holding this up.  LGTM
